### PR TITLE
Formatter Considers comments

### LINKED ISSUE #74: Comments are not considered in Formatting
The XMLScanner is setup for comments, but it is not handled in XMLFormatting, formatting will remove the comments. 

To fix add consideration of comments in XMLParser

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Fixes #74

### CODE REVIEW

**Comment by fbricon** [comment]
```
Failed tests:   testHTMLAttributeNameCompletion(org.eclipse.lsp4xml.emmet.participants.EmmetCompletionTest): a>b[c] should only existing once: Actual:  expected:<1> but was:<0>
  testHTMLElementCompletion(org.eclipse.lsp4xml.emmet.participants.EmmetCompletionTest): a>b should only existing once: Actual:  expected:<1> but was:<0>

Tests in error:
  testExtract(org.eclipse.lsp4xml.emmet.emmetio.extract.ExtractAbbreviationTest)

```

**Comment by fbricon** [comment]
actually I get the same test failures on master. cc @angelozerr

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/internal/parser/XMLParser.java [hunk 1]
```diff
@@ -24,7 +27,7 @@
  *
  */
 public class XMLParser {
-
+	private static final Logger LOGGER = Logger.getLogger(XMLParser.class.getName());
```
Comment (angelozerr) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/internal/parser/XMLParser.java]: Please remove this unused field.

**Comment by angelozerr** [comment]
@NikolasKomonen please resolve conflict. Thanks!

**Comment by angelozerr** [comment]
> actually I get the same test failures on master. cc @angelozerr

It should be fixed now.

**Comment by fbricon** [comment]
Considering :
```xml
<foo>
    <bar>hi</bar> <!-- important comment -->
</foo>
```
formatting yields:
```xml
<foo>
	<bar>hi</bar>
	<!-- important comment -->
</foo>
```
Comments should not be moved below the tag they're next to

**Comment by NikolasKomonen** [comment]
Is this fine?
```xml
<foo>
    <bar><!-- important comment -->hi</bar> 
</foo>
```
formatting yields:
```xml
<foo>
	<bar>hi</bar>
	<!-- important comment -->
</foo>
```
Comments should not be moved below the tag they're next to

**Pull request merged**